src/main/java/org/jahia/services/usermanager/ldap/LDAPUserGroupProvider.java

```diff
@@ -709,41 +705,30 @@ private LDAPGroupCacheEntry getGroupCacheEntry(String groupName, boolean cache)
      * @param isDynamic
      * @return
      */
-    private LDAPGroupCacheEntry getGroupCacheEntryByName(final String name, boolean cache, final boolean isDynamic) throws Exception {
+    private LDAPGroupCacheEntry getGroupCacheEntryByName(final String name, final boolean cache, final boolean isDynamic) throws Exception {
 
         final List<String> groupAttrs = getGroupAttributes(isDynamic);
-        final GroupNameClassPairCallbackHandler nameClassPairCallbackHandler = new GroupNameClassPairCallbackHandler(null, isDynamic);
         long startTime = System.currentTimeMillis();
-        final Exception[] exceptions = new Exception[1];
-
-        boolean validLdapCall = ldapTemplateWrapper.execute(new BaseLdapActionCallback<Boolean>(getExternalUserGroupService(), getKey()) {
+        
+        final LDAPGroupCacheEntry groupCacheEntry = ldapTemplateWrapper.execute(new BaseLdapActionCallback<LDAPGroupCacheEntry>(getExternalUserGroupService(), getKey()) {
 
             @Override
-            public Boolean doInLdap(LdapTemplate ldapTemplate) {
+            public LDAPGroupCacheEntry doInLdap(LdapTemplate ldapTemplate) {
+                final GroupNameClassPairCallbackHandler nameClassPairCallbackHandler = new GroupNameClassPairCallbackHandler(null, isDynamic);
                 ldapTemplate.search(applyPredefinedGroupFilter(query().base(groupConfig.getSearchName())
-                                .attributes(groupAttrs.toArray(new String[groupAttrs.size()]))
-                                .where(OBJECTCLASS_ATTRIBUTE).is(isDynamic ? groupConfig.getDynamicSearchObjectclass() : groupConfig.getSearchObjectclass())
-                                .and(groupConfig.getSearchAttribute()).is(decode(name))),
+                        .attributes(groupAttrs.toArray(new String[groupAttrs.size()]))
+                        .where(OBJECTCLASS_ATTRIBUTE).is(isDynamic ? groupConfig.getDynamicSearchObjectclass() : groupConfig.getSearchObjectclass())
+                        .and(groupConfig.getSearchAttribute()).is(decode(name))),
                         nameClassPairCallbackHandler);
-                return true;
-            }
-
-            @Override
-            public Boolean onError(Exception e) {
-                exceptions[0] = e;
-                super.onError(e);
-                return false;
+                return getAndCacheGroupEntry(nameClassPairCallbackHandler, cache);
             }
         });
 
-        if (!validLdapCall) {
-            throw exceptions[0];
-        }
         if (logger.isDebugEnabled()) {
             logger.debug("Get group {} in {} ms", name, System.currentTimeMillis() - startTime);
         }
 
-        return getAndCacheGroupEntry(nameClassPairCallbackHandler, cache);
+        return groupCacheEntry;
     }
```

Comment:
As far as I understand, if getGroupCacheEntryByName() returns null instead of throwing an exception, as it seems to be the case with the fix, the group will be cached as non existing - which do not fulfil "In the same way, when an error occurs, we should not cache the incomplete results or at least not the same amount of time.",  and also break the previous behaviour ( which was also listed as https://jira.jahia.org/browse/QA-8400 ).
Something must be done to not cache this information ?

Reply:
Indeed, the modifications for the function "getGroupCacheEntryByName" can be reversed.